src/server/pkg/pool/pool.go

```diff
@@ -109,7 +109,7 @@ func (p *Pool) Do(ctx context.Context, f func(cc *grpc.ClientConn) error) error
 		for {
 			for addr, mapConn := range p.conns {
 				if mapConn.cc == nil {
-					cc, err := grpc.DialContext(ctx, addr, p.opts...)
+					cc, err := grpc.DialContext(ctx, "dns:///"+addr, p.opts...)
```

Comment:
I don't think we actually want it here, since the new DNS resolution behavior should be client-side only

Reply:
I don't have a strong opinion on this. But the reason to go change all the grpc.Dial call is so that we have the same grpc options used everywhere. I don't think we have any specific DNS configuration dependency.

If we think this is can have unintended side effects, I'll remove it.

Reply:
It's not clear to me yet how `dns:///` changes behavior. gRPC apparently [has a system for plugging in resolvers](https://godoc.org/google.golang.org/grpc#WithResolvers), so my off-hand guess (without having dug too much into the code) is that using the dns uri causes gRPC to use one of its own resolvers, which follows proper system behavior, rather than go stdlib's, which [hand-rolled its own janky solution when cgo is disabled.](https://golang.org/pkg/net/#hdr-Name_Resolution)

My concern with bringing this change to the server-side is that we don't really know what happens when you use the `dns` uri though. What we do know is that:

1) DNS resolution is broken for us client-side on macOS.
2) We haven't had any reports of issues server-side. 
3) Even with these changes, server-side DNS is going to be radically different because it's running on linux, and because it's on kubernetes (which messes around with DNS.)

By restricting these changes to the client-side, we run the risk of "just" breaking `pachctl`.

Reply:
@ysimonson I made the change to be client-side only. It's going through CI. Please take a look.